### DISCUSSION TITLE
Github Actions - Can't connect to mysql

### DISCUSSION THREAD

**Discussion by sidd9433** [community/community#25200] [Actions]
<p>Hi All,</p>
<p>I am trying to connect to the mysql database from ly pipeline, but its giving error. Although it used to work.</p>
<p>Part of my pipeline.yml</p>
<pre><code>jobs:
  build-and-publish:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - name: Set up JDK 11
      uses: actions/setup-java@v1
      with:
        java-version: 11
    - name: Create test database
      run: |
        mysql -uroot -proot &lt; databases/tracker/create_databases.sql
        ./gradlew testMigrate
    - name: Build with Gradle
      run: ./gradlew build
    - name: Publish artifact
      uses: actions/upload-artifact@v1
</code></pre>
<p>Error message:</p>
<pre><code>Run mysql -uroot -proot &lt; databases/tracker/create_databases.sql
  mysql -uroot -proot &lt; databases/tracker/create_databases.sql
  ./gradlew testMigrate
  shell: /bin/bash -e {0}
  env:
    JAVA_HOME: /opt/hostedtoolcache/jdk/11.0.6/x64
    JAVA_HOME_11.0.6_x64: /opt/hostedtoolcache/jdk/11.0.6/x64
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
##[error]Process completed with exit code 1.
</code></pre>
<p>I have tried with </p>
<pre><code>mysql -h127.0.0.1 -uroot -proot &lt; databases/tracker/create_databases.sql
</code></pre>
<p>Error message:</p>
<pre><code>Create test database4s
##[error]Process completed with exit code 1.
Run mysql -uroot -proot -h127.0.0.1 &lt; databases/tracker/create_databases.sql
  mysql -uroot -proot -h127.0.0.1 &lt; databases/tracker/create_databases.sql
  ./gradlew testMigrate
  shell: /bin/bash -e {0}
  env:
    JAVA_HOME: /opt/hostedtoolcache/jdk/11.0.6/x64
    JAVA_HOME_11.0.6_x64: /opt/hostedtoolcache/jdk/11.0.6/x64
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1' (111)
##[error]Process completed with exit code 1.
 Build with Gradle
</code></pre>
<p>Can anyone help. Has anything changed? Am I doing anything wrong?</p>
<p>Thanks,</p>
<p>Sid</p>

**Comment by Seth10001** [comment]
<p>I’m having this issue too. I think the <strong>ubuntu-latest</strong> image might have changed?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>The issue is caused by this change: <a href="https://github.blog/changelog/2020-02-21-github-actions-breaking-change-ubuntu-virtual-environments-will-no-longer-start-the-mysql-service-automatically/" rel="nofollow noopener">https://github.blog/changelog/2020-02-21-github-actions-breaking-change-ubuntu-virtual-environments-will-no-longer-start-the-mysql-service-automatically/</a></p>
<p>You could start mysql before creating database:  </p>
<pre><code>- name: Create test database
   run: |
**sudo systemctl start mysql** mysql -uroot -proot &lt; databases/tracker/create_databases.sql
</code></pre>

**Reply by aroraank11** [reply]
I was facing the same issue then I tried running following 

```
- name: Importing MYSQL file
      env:
        DB_HOST: 127.0.0.1
        DB_CONNECTION: mysql
        DB_DATABASE: test
        DB_PORT: ${{ job.services.mysql.ports[3306] }}
        DB_USER: root
        DB_PASSWORD: password
      run: |
          sudo /etc/init.d/mysql start
          mysql -Dtest -uroot -ppassword -hlocalhost -P33306 < "api_token.sql"
```
so now error code has changed 

```
Starting mysql (via systemctl): mysql.service.
Warning: arning] Using a password on the command line interface can be insecure.
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
Error: Process completed with exit code 1.
```

any help would be appreciated. Thanks

**Reply by bkestelman** [reply]
The default password is root, not password

**Reply by k00ni** [reply]
Even though this post was marked as answer, there is a small detail which might cause trouble when the code is just copied.

In the line:

>           mysql -Dtest -uroot -ppassword -hlocalhost -P33306 < "api_token.sql"

the port is `33306` and not `3306`. The latter is the default port, so using `33306` might cause further problems.
